docs/root/intro/life_of_a_request.rst

```diff
@@ -0,0 +1,181 @@
+applications. In the service mesh model:
+
+* Ingress listeners take requests from other nodes in the service mesh and forward them to the
+  local application. Responses from the local application flow back through Envoy to the other
+  nodes in the service mesh.
+* Egress listeners take requests from the local application and forward them to other nodes in the
+  network. These nodes will also be typically running Envoy and accepting the request via their
+  ingress listeners.
+
+.. note::
+  TODO: add a service mesh diagram
+
+Envoy is used in a variety of domains beyond the service mesh. For example, it can also act
+as an internal load balancer and as an edge networking ingress/egress proxy.
+
+.. note::
+  TODO: add topology diagrams for edge and ILB.
+
+Envoy may be configured in multi-tier topologies for scalability and reliability:
+
+.. note::
+  TODO: add topology diagrams for multi-tier.
+
+In all the above cases, a request will arrive at a specific Envoy via TCP, UDP or Unix Domain
+Sockets from a downstream. Envoy will forward requests to upstreams via TCP or Unix Domain
+Sockets. We focus on a single Envoy proxy that is part of one of these network topologies
+below.
+
+Configuration
+-------------
+
+Envoy is a platform with a high degreee of extensibility. This results in a combinatorial
+explosion of possible request paths, depending on:
+
+* L3/4 protocol, e.g. TCP, UDP, Unix Domain Sockets.
+* L7 protocol, e.g. HTTP/1, HTTP/2, HTTP/3, gRPC, Thrift, Dubbo, Kafka, Redis and various databases.
+* Transport socket, e.g. plain text, TLS, ALTS.
+* Routing, e.g. PROXY protocol, original destination, dynamic forwarding.
+* Authentication and authorization.
+* Circuit breakers and outlier detection.
+* Many other networking, HTTP, listener, access logging, health checking, tracing and stats
+  extensions and configuration options.
+
+It's helpful to limit the nature of the configuration and request to provide an illustrative
+narrative below, so we opt for the following parameters:
+
+* An HTTP/2 request with TLS over a TCP connection for both downstream and upstream.
+* The HTTP connection manager as the only network filter.
+* The router filter as the only HTTP filter (TODO: should we add more filters?).
+* Filesystem access logging.
+* Statsd sink.
+
+We assume a static bootstrap configuration, but this is a control plane detail that is
+orthogonal to the data plane:
+
+.. code-block:: yaml
+
+  todo: simple bootstrap example of above
+
+High level architecture
+-----------------------
+
+The request processing path in Envoy has responsibilities split across two major loosely
+coupled subsystems:
+
+* *ListenerManager* where the downstream client related aspects of request processing take place.
+  This includes all processing from when the first bytes are read from the socket until a
+  decoded HTTP request is ready to forward to an upstream cluster. This subsystem is also
```

Comment:
upstream cluster

hm, per my point about downstream and upstream, maybe even worth defining some terms at the top?  Or links to definitions? Especially for interns, they may not be familiar with cluster, connection pooling, etc.

Apart from the language pass we could ask someone not proxy savvy to take a read through and tell us what's not obvious, since we've been immersed in this space long enough we're probably going to miss a bunch.  Given the more clear we make this guide, the easier our onboarding is, I think it's worth getting a pass either from a noogler in our org, or some coworkers who aren't doing proxying, so we can catch as many of those before summer as we can.

Reply:
I agree; I have some folks in mind for this, so will ping them once we have an approximately complete doc.